NullTerminatorsTeamProject/src/main/java/com/nullterminators/project/controller/EmployeeProfileController.java

```diff
@@ -0,0 +1,114 @@
+package com.nullterminators.project.controller;
+
+import com.nullterminators.project.model.EmployeeProfile;
+import com.nullterminators.project.service.EmployeeProfileService;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * API endpoints for employee profile management.
+ */
+@RestController
+@RequestMapping("/employeeProfile")
+public class EmployeeProfileController {
+    
+  @Autowired
+  private EmployeeProfileService employeeProfileService;
+
+  /**
+   * GET /employeeProfile/ - gets a list of all existing employee profiles.
+   *
+   * @return ResponseEntity with appropriate status and message
+   */
+  @GetMapping({"/", "/getAllEmployees"})
+  public ResponseEntity<?> getAllEmployees() {
+    try {
+      List<EmployeeProfile> allEmployees = employeeProfileService.getAllEmployees();
+      return new ResponseEntity<>(
+          allEmployees, HttpStatus.OK);
+    } catch (Exception e) {
+      return handleException(e);
+    }
+  }
+
+  /**
+   * GET /employeeProfile/{id} - gets an employee profile by id.
+   *
+   * @param id (int) : employee id
+   * @return ResponseEntity with appropriate status and message
+   */
+  @GetMapping("/{id}")
+  public ResponseEntity<?> getEmployee(@PathVariable int id) {
+    try {
+      Optional<EmployeeProfile> employee = employeeProfileService.getEmployeeProfile(id);
+      if (employee.isPresent()) {
+        return new ResponseEntity<>(employee.get(), HttpStatus.OK);
+      } else {
+        return new ResponseEntity<>("Employee with id " + id + " does not exist.",
+          HttpStatus.NOT_FOUND);
+      }
+    } catch (Exception e) {
+      return handleException(e);
+    }
+  }
+
+  /**
+   * POST /employeeProfile/createNewEmployee - creates new employee profile.
+   *
+   * @param employeeProfile (EmployeeProfile)
+   * @return ResponseEntity with appropriate status and message
+   */
+  @PostMapping("/createNewEmployee")
+  public ResponseEntity<?> createNewEmployee(@RequestBody EmployeeProfile employeeProfile) {
```

Comment:
When creating a profile, are you taking ID also from input or are you generating it?

Reply:
Generating it